test/Microsoft.ML.CpuMath.UnitTests.netstandard/UnitTests.cs

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.ML.Runtime.Internal.CpuMath;
+
+namespace Microsoft.ML.CpuMath.UnitTests
+{
+    public class CpuMathUtilsUnitTests
+    {
+        private readonly float[][] testArrays;
+        private readonly int[] testIndexArray;
+        private const float DEFAULT_SCALE = 1.7f;
+        private FloatEqualityComparer comparer;
+
+        public CpuMathUtilsUnitTests()
+        {
+            // padded array whose length is a multiple of 4
+            float[] testArray1 = new float[8] { 1.96f, -2.38f, -9.76f, 13.84f, -106.37f, -26.93f, 32.45f, 3.29f };
+            // unpadded array whose length is not a multiple of 4.
+            float[] testArray2 = new float[7] { 1.96f, -2.38f, -9.76f, 13.84f, -106.37f, -26.93f, 32.45f };
+            testArrays = new float[][] { testArray1, testArray2 };
+            testIndexArray = new int[4] { 0, 2, 5, 6 };
+            comparer = new FloatEqualityComparer();
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public unsafe void DotUTest(int test)
+        {
+            float[] src = (float[]) testArrays[test].Clone();
+            float[] dst = (float[]) src.Clone();
+            
+            for (int i = 0; i < dst.Length; i++)
+            {
+                dst[i] += 1;
+            }
+
+            var managedOutput = CpuMathUtils.DotProductDense(src, dst, dst.Length);
+
+            if (test == 0)
+            {
+                Assert.Equal(13306.0376f, managedOutput, 4);
+            }
+            else
+            {
+                Assert.Equal(13291.9235f, managedOutput, 2);
+            }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public unsafe void DotSUTest(int test)
+        {
+            float[] src = (float[])testArrays[test].Clone();
+            float[] dst = (float[])src.Clone();
+            int[] idx = testIndexArray;
+
+            for (int i = 0; i < dst.Length; i++)
+            {
+                dst[i] += 1;
```

Comment:
Why are we doing this?  (repeated else where)

Reply:
Wanted to test the correctness of the dot product of two unequal float arrays.  Without `dst[i] += 1`, `src` and `dst` are the same, and the dot product result may look like that from the `SumSqU` function.  But I agree that it is not entirely necessary.

Reply:
I think that is a fine reason. It would be good to stick a code comment in here stating that. Then others reading the code know why this is here, and don't delete it.